### ISSUE TITLE
Feature request: add support for ES6 Sets

### ISSUE THREAD

**Issue by AndiDog** [kolodny/immutability-helper#63]
I'd love to see `$add`/`$delete` being added to the library.

Use case example is a React table component which has `state.expandedRows: Set<string>` where each item of the set represents a row which has some expanded extra information. Using a `Set` is much nicer to read than abusing `$set`/`$unset` for this purpose (which I'll try right now as workaround).

**Comment by kolodny** [comment]
Hmm, yeah sets and maps shouldn't be broken. I need to look into how to fix them. I suspect it's a bad `copy` implementation.

That said (and once it's fixed), adding something to a set is actually a function call and not a simple assignment operation. I would suggest using `$apply` to accomplish that:


```js
const s = { a: Set(['a']) }
const s2 = update(s, {a: { $apply: newSet => newSet.add('thing') }})
```

**Comment by rob2d** [comment]
If you need help and have had any insights into it, let me know. I have some time coming up and this is an especially useful library for me as I don't always need the bells and whistles of ImmutableJS and it's awesome on smaller projects :)

**Comment by kolodny** [comment]
Sorry for the delay, This has been done and published to `v2.5.0`.
@rob2d Sorry for the lack of response, I didn't get around to seeing this for a while and figured you didn't have the bandwidth anymore. Thanks for the offer!

**Comment by mnpenner** [comment]
It doesn't appear to be documented, so I'll mention it here: this has been implemented as `$add` and `$remove` *not* `$add` and `$delete` as OP originally wrote.

These are great BTW. Came in handy already 😄 

Also, they appear to work on `Map`s as well.

**Comment by kolodny** [comment]
Sorry about that, I just published those changes to 2.5.1

I used $remove and not $delete as I thought that would be confused with the `delete` operator.
